apps/src/code-studio/components/libraries/LibraryPublisher.jsx

```diff
@@ -119,8 +121,17 @@ export default class LibraryPublisher extends React.Component {
       libraryJson,
       error => {
         console.warn(`Error publishing library: ${error}`);
+
         if (error.message.includes('httpStatusCode: 413')) {
           this.setState({publishState: PublishState.TOO_LONG});
+        } else if (error.message.includes('ShareFailure')) {
+          let match = error.message.match(/ShareFailure: content=(.+)/);
+          if (match && match.length > 1) {
+            this.setState({
+              publishState: PublishState.PII_INPUT,
+              pIIWord: match[1],
```

Comment:
The ShareFiltering logic will return after the first instance of PII, even if there are multiple offenses.

Reply:
Note: I modified this in a follow-up commit so that `pIIWords` is an array of 1 (to be consistent with the handling of `profaneWords`.